libs/client/my-stuff/src/lib/components/upload-cover-art/upload-cover-art.component.ts

```diff
@@ -128,16 +135,16 @@ export class UploadCoverArtComponent {
         this.dialogRef.close();
     }
 
-    uploadCoverArt() {
-        const token = localStorage.getItem('auth.token');
-        this.uploader.authToken = `Bearer ${token}`;
+    async uploadCoverArt() {
+        this.uploader.authToken = `Bearer ${this.token}`;
         this.loading = true;
         this.uploader.clearQueue();
         this.uploader.addToQueue([this.fileToReturn]);
 
-        this.stuff.uploadCoverArt(this.uploader);
-        this.loading = false;
-        this.dialogRef.close();
+        await this.stuff.uploadCoverArt(this.uploader).then(() => {
+            this.loading = false;
+            this.dialogRef.close();
+        });
```

Comment:
Figs what have I told you about .`then()`

_you are even using `await` what are you doing_

Reply:
oh that's so the dialog doesn't close until the upload is successful

Reply:
actually, the more I read this, the less sense it makes. please hold while I pull this locally